__tests__/app.test.js

```diff
@@ -351,9 +351,49 @@ describe("GET /api/users",()=>{
           expect(user).toMatchObject({
             username: expect.any(String),
             name: expect.any(String),
-            avatar_url: expect.any(String)
+            avatar_url: expect.any(String),
           });
         });
       });
   });
-})
+});
+
+describe("GET /api/articles (topic query)", () => {
+  test("GET200: Endpoint accept topic query and responds articles with seleted topics", () => {
+    return request(app)
+      .get("/api/articles?topic=cats")
+      .expect(200)
+      .then(({ body: { allArticles } }) => {
+        expect(allArticles.length).toBe(1);
+        allArticles.forEach((article) => {
+          expect(article).toMatchObject({
+            title: expect.any(String),
+            topic: "cats",
+            author: expect.any(String),
+            body: expect.any(String),
+            created_at: expect.toBeDateString(),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(Number),
+          });
+        });
+      });
+  });
+  test("GET404: Respond with an error when passed non-existent topic", () => {
+    return request(app)
+      .get("/api/articles?topic=abc")
+      .expect(404)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("Non-existent Topic");
+      });
+  });
+  test("GET404: Respond with an error when invalid query", () => {
```

Comment:
this should maaaybe be a 400 instead since it's an invalid request

Reply:
Oops, it is 400 in the test, and should be GET400 in the description 😅 just updated it. 

I do struggle to decide to go for 400 or 404 every time, is there like a way to know which one I should use? Or is it different for every situation?